processing/src/main/java/org/apache/carbondata/processing/datatypes/ArrayDataType.java

```diff
@@ -183,7 +187,27 @@ public void parseAndBitPack(ByteBuffer byteArrayInput, DataOutputStream dataOutp
     for (int i = 0; i < dataLength; i++) {
       children.parseAndBitPack(byteArrayInput, dataOutputStream, generator);
     }
+  }
+
+  @Override
+  public int parseComplexValue(ByteBuffer byteArrayInput, DataOutputStream dataOutputStream,
+      KeyGenerator[] generator, Boolean[][] complexDictionaryIndentification, int startOffset)
+      throws IOException, KeyGenException {
+    int dataLength = byteArrayInput.getInt();
+    startOffset += Integer.SIZE / Byte.SIZE;
```

Comment:
No need to calculate every time , just use `4` and add comment

Reply:
Added CarbonCommonConstant